app/layouts/DashboardLayout/OnboardingModal.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import type { FC } from 'react';
+
+import useFullService from '../../hooks/useFullService';
+import SetPinModal from './SetPinModal';
+import ShowEntropyModal from './ShowEntropyModal';
+
+const OnboardingModal: FC = () => {
+  const {
+    confirmEntropyKnown,
+    isEntropyKnown,
+    isPinRequired,
+    pendingSecrets,
+    setPinWithoutPassword,
+  } = useFullService();
```

Comment:
In the proposed refactor we'd actually be making these calls in the DashboardLayout and passing down to OnboardingModal correct?

Reply:
Yes, I would expect them to be at the top-most layer. So either Views or Layouts.